lib/matplotlib/figure.py

```diff
@@ -266,11 +266,9 @@ def __str__(self):
         return "Figure(%gx%g)" % tuple(self.bbox.size)
 
     def __repr__(self):
-        return "<{clsname} size {h:g}x{w:g} with {naxes} Axes>".format(
-            clsname=self.__class__.__name__,
-            h=self.bbox.size[0], w=self.bbox.size[1],
-            naxes=len(self.axes),
-        )
+        return (f"<{self.__class__.__name__} size {self.bbox.size[0]:g}"
+                f"x{self.bbox.size[1]:g} with {len(self.axes)} Axes; "
+                f"0x{id(self):x}>")
```

Comment:
Default reprs read like `<__main__.Foo instance at 0x2afd73c49518>`. Maybe we should also use `at 0x...` to be more consistent the the standard repr.